pkg/datapath/linux/requirements.go

```diff
@@ -135,6 +135,12 @@ func CheckMinRequirements() {
 	// bpftool checks
 	if !option.Config.DryMode {
 		probeManager := probes.NewProbeManager()
+
+		// Warn about kernel not supporting large 1M instruction
+		if err := probeManager.ProbeKernelLargeInstruction(); err != nil {
+			log.WithError(err).Warn("BPF check: NOT OK")
+		}
```

Comment:
I think we need to consider exactly what we're trying to communicate here.

As-is, this code will *always* warn any user running roughly version ~5.2 of Linux or earlier. There is no impact to those users, it's just noise. Many distributions rely on Linux 4.14, 4.15, 4.19. If users see this warning, those users on those kernels may be under the impression that something is broken, but nothing is broken. Also, what is the user supposed to do about this? `BPF check: NOT OK` is pretty cryptic as a message.

I'm not sure, do we need this particular hunk of code at all?

Reply:
@joestringer This is to address https://github.com/cilium/cilium/pull/17370 exceeding 4.19 kernel  4K instruction limit, I thought you mentioned to send a separate PR for kernel limit probing and warning https://github.com/cilium/cilium/pull/17370#issuecomment-932579022, or I misunderstand you :)

Reply:
I see what you mean, I can  warn only when https://github.com/cilium/cilium/pull/17370 is enabled. for the same context, I have https://github.com/cilium/cilium/pull/17370#issuecomment-934699604 that there is really some problem between Cilium and Kernel 4.9 - 5.4, I think my PR simply just exposed the problem, appreciate you can  provide feed back